test/compiler_test.js

```diff
@@ -87,14 +93,46 @@ describe('Compiler Tests', () => {
                 .setGlobal(payload);
 
               // eslint-disable-next-line import/no-dynamic-require,global-require
-              const service = require(copiledFilename);
+              const service = require(compiledFilename);
               service(runtime).then(() => {
                 const output = runtime.stream;
                 assert.equal(output, test.output);
                 done();
               }).catch(done);
             });
           }
+
+          if ('mappedOutput' in test) {
+            const mapFilename = `${compiledFilename}.map`;
+            it(`${idx}. Maps lines for '${test.name}' correctly.`, (done) => {
+              fs.readFile(mapFilename, 'utf8', (err1, map) => {
+                if (err1) {
+                  throw err1;
+                }
+                fs.readFile(compiledFilename, 'utf8', (err2, js) => {
```

Comment:
use `fs.readFileSync()`. makes it easier to read.

Reply:
it's good practice to avoid sync methods whenever possible. 

[fs-extra](https://github.com/jprichardson/node-fs-extra) is a drop-in replacement for the builtin `fs` module and provides convenient async/await and Promise-based signatures, e.g.

```javascript
const js = await fs.readFile(compiledFilename, 'utf8'); 
```

Reply:
We use `fs-extra` almost consistently everywhere across the code base, so it makes sense to use it here, too.